src/components/CourseInfo/CourseInfo.jsx

```diff
@@ -0,0 +1,80 @@
+import { useEffect } from 'react';
+import { Link, useParams, useNavigate } from 'react-router-dom';
+import PropTypes from 'prop-types';
+
+import { pipeDuration } from '../../helpers/pipeDuration';
+import { dateGenerator } from '../../helpers/dateGeneratop';
+
+/**
+ * Course Info React component
+ */
+function CourseInfo(props) {
+	const params = useParams();
+	const navigate = useNavigate();
+	const courseId = params.courseId;
+
+	const course = props.coursesList.find((item) => item.id === courseId);
+
+	// If course is not exist, redirect to the Courses List
+	useEffect(() => {
+		if (!course) {
+			navigate('/courses');
+		}
+	});
```

Comment:
Same here, you run it every render, but my guess you only need it to run on component mount